modules/installation-aws-upi-installation.adoc

```diff
@@ -0,0 +1,164 @@
+test-tkh7l-worker-us-east-2c   1         1                               11m
+----
++
+Note the `NAME` of each MachineSet. Because you use a different subnet than the
+installation program expects, the worker MachineSets do not use the correct
+network settings. You must edit each of these MachineSets.
+
+. Edit each worker MachineSet to provide the correct values for your cluster:
++
+----
+$ oc edit machineset --namespace openshift-machine-api test-tkh7l-worker-us-east-2a -o yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  creationTimestamp: 2019-03-14T14:03:03Z
+  generation: 1
+  labels:
+    machine.openshift.io/cluster-api-cluster: test-tkh7l
+    machine.openshift.io/cluster-api-machine-role: worker
+    machine.openshift.io/cluster-api-machine-type: worker
+  name: test-tkh7l-worker-us-east-2a
+  namespace: openshift-machine-api
+  resourceVersion: "2350"
+  selfLink: /apis/machine.openshift.io/v1beta1/namespaces/openshift-machine-api/machinesets/test-tkh7l-worker-us-east-2a
+  uid: e2a6c8a6-4661-11e9-a9b0-0296069fd3a2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: test-tkh7l
+      machine.openshift.io/cluster-api-machineset: test-tkh7l-worker-us-east-2a
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: test-tkh7l
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: test-tkh7l-worker-us-east-2a
+    spec:
+      metadata:
+        creationTimestamp: null
+      providerSpec:
+        value:
+          ami:
+            id: ami-0eecbb884c8b35b1e <1>
+          apiVersion: awsproviderconfig.openshift.io/v1beta1
+          blockDevices:
+          - ebs:
+              iops: 0
+              volumeSize: 120
+              volumeType: gp2
+          credentialsSecret:
+            name: aws-cloud-credentials
+          deviceIndex: 0
+          iamInstanceProfile:
+            id: test-tkh7l-worker-profile
+          instanceType: m4.large
+          kind: AWSMachineProviderConfig
+          metadata:
+            creationTimestamp: null
+          placement:
+            availabilityZone: us-east-2a
+            region: us-east-2
+          publicIp: null
+          securityGroups: <2>
+          - filters:
+            - name: tag:Name
+              values:
+              - test-tkh7l-worker-sg
+          subnet: <3>
+            filters:
+            - name: tag:Name
+              values:
+              - test-tkh7l-private-us-east-2a
+          tags:
+          - name: kubernetes.io/cluster/test-tkh7l
+            value: owned
+          userDataSecret:
+            name: worker-user-data
+      versions:
+        kubelet: ""
+status:
+  fullyLabeledReplicas: 1
+  observedGeneration: 1
+  replicas: 1
+----
+<1> I think you change the RHCOS AMI here.
```

Comment:
@cuppett, do you know if there's a good RHCOS AMI that we should use?

Do you have more details about the values that a user would need to provide to update these MachineSets?

Reply:
The IAM instance profile name comes as an output from the 03_cluster_security.yaml file. The subnet is created as part of 01_vpc.yaml; however, it doesn't have a name tag to filter on. They can change the specification to ID and use the exact ID or add a name. We could also add a generated name tag to help minimize the edit needed. The same is true for the security group.

For the AMI, we build a different one each day. I usually just query the AWS console to grab a recent one. @wking may have a more prescriptive method.

Reply:
> For the AMI...

See [here][1] and [here][2] for scripted approaches to figuring out your AMI.  And see [here][3] for possible future work.

[1]: https://github.com/openshift/installer/pull/1423#issuecomment-479348663
[2]: https://github.com/openshift/installer/pull/1168#issuecomment-459501811
[3]: https://github.com/openshift/installer/issues/1399

Reply:
Thank you! I'll watch the RFE because that will be useful.